libcxx/include/__system_error/errc.h

```diff
@@ -113,6 +113,15 @@ _LIBCPP_BEGIN_NAMESPACE_STD
 // for them:
 
 // enum class errc
+//
+// LWG3869 deprecates the UNIX STREAMS macros and enum values.
+// This makes the code clumbersome:
+// - the enum value is deprecated and should show a diagnostic,
+// - the macro is deprecated and should _not_ show a diagnostic in this
+//   context, and
+// - the macro is not always available.
+// This leads to the odd pushing and popping of the deprecated
```

Comment:
If a declaration itself is deprecated and uses deprecated functionality, the deprecation warning of the stuff it uses should be suppressed: https://godbolt.org/z/b839bqGj7. This looks like a Clang bug to me. Could you file one?

Reply:
@AaronBallman do you agree it's a bug? If so I'm happy to file a report.

Reply:
Oofda. I want to call this a bug, but it's going to be exceptionally hard to work around because of the difference in phases of translation. We issue the deprecated macro warning when the macro is expanded, which is *long* before we have any idea of how the value is being used. So it may be worth filing a QoI enhancement request over it, but I don't know that we have a way to improve it just yet.

Reply:
Thanks I've filed https://github.com/llvm/llvm-project/issues/81359